internal/dataplane/parser/translators/grpcroute_atc.go

```diff
@@ -29,6 +29,32 @@ func GenerateKongExpressionRoutesFromGRPCRouteRule(grpcroute *gatewayv1alpha2.GR
 	// gather the k8s object information and hostnames from the grpcroute
 	ingressObjectInfo := util.FromK8sObject(grpcroute)
 
+	// generate a route to match hostnames only if there is no match in the rule.
+	if len(rule.Matches) == 0 {
+		// REVIEW: return an error here to tell parser register a translation error event?
+		if len(grpcroute.Spec.Hostnames) == 0 {
+			return nil
+		}
```

Comment:
Ditto the traditional handler. I think these do result in routes that match any hostname.

Reply:
Ditto here.